### PR TITLE
feat(utils): define var as global var

### PR DESCRIPTION
**Pull request by PablitoGS**
we need to move this var out of scope to redefine our black, by default we don't use pure black in our theme.

### CODE REVIEW

File: src/utils/_colors.scss [hunk 1]
```diff
@@ -12,10 +12,16 @@
...
+$color-variation-dark: black;
 $color-lighten-steps: 25, 50, 75, 90, 95 !default;
 $color-darken-steps: 20, 35, 55, 75 !default;
 
-@function color-variation($color, $step: 0, $dark: black, $light: white) {
+@function color-variation(
+  $color,
+  $step: 0,
+  $color-variation-dark,
+  $light: white
```
Comment (turolopezsanabria) [inline src/utils/_colors.scss:23]: wouldn't you change this name to "$color-variation-light" as well?

File: src/utils/_colors.scss [hunk 2]
```diff
@@ -12,10 +12,16 @@
 // $c-primary-dark: color-variation($c-primary, -2) -> 2 steps darker
 // $c-primary-darkest: color-variation($c-primary, -4) -> 4 steps darker
 
+$color-variation-dark: black;
```
Comment (turolopezsanabria) [inline src/utils/_colors.scss:15]: I think you need a  ! default here to be able to customise in your theme

File: src/utils/_colors.scss (hunk 1 above)
Comment (PablitoGS) [reply src/utils/_colors.scss]: yes! done

File: src/utils/_colors.scss (hunk 2 above)
Comment (PablitoGS) [reply src/utils/_colors.scss]: ups, right, done!

File: src/utils/_colors.scss [hunk 3]
```diff
@@ -12,23 +12,47 @@
...
 $color-lighten-steps: 25, 50, 75, 90, 95 !default;
 $color-darken-steps: 20, 35, 55, 75 !default;
 
-@function color-variation($color, $step: 0, $dark: black, $light: white) {
+@function color-variation(
+  $color,
+  $step: 0,
+  $color-variation-dark,
+  $color-variation-light
+) {
```
Comment (nguasch) [inline src/utils/_colors.scss:26]: you need to set `$color-variation-dark` and `$color-variation-light` with global vars (`$color-variation-dark` and `$color-variation-light`, maybe this two with an other name)

Comment (PablitoGS) [reply src/utils/_colors.scss:26]: done

**Comment by PablitoGS** [comment]
@turolopezsanabria now $dark have var value and I understand it doesn't break what we have

**Pull request merged**